marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/core.sjs

```diff
@@ -36,14 +36,16 @@ function getTypesInfo() {
     for (const artifactType of Object.keys(registeredArtifactTypes)) {
         if (registeredArtifactTypes.hasOwnProperty(artifactType)) {
             const artifactLibrary = registeredArtifactTypes[artifactType];
+            const updateRoles = artifactLibrary.getPermissions().filter((perm) => perm.capability === 'update').map((perm) => perm.roleId);
+            const currentRoles = xdmp.getCurrentRoles().toArray();
+            const userCanUpdate = updateRoles.some((roleId) => currentRoles.includes(roleId));
             typesInfo.push({
                 type: artifactType,
-                storageDatabases: artifactLibrary.getStorageDatabases(),
-                collections: artifactLibrary.getCollections(),
                 fileExtension: getArtifactFileExtension(artifactType),
                 directory: getArtifactDirectory(artifactType),
                 nameProperty: artifactLibrary.getNameProperty(),
-                versionProperty: artifactLibrary.getVersionProperty()
+                versionProperty: artifactLibrary.getVersionProperty(),
+                userCanUpdate: userCanUpdate
```

Comment:
@ryanjdew I just tested it, userCanUpdate is always false. After I did the above changes as suggestion, it works.
Also can we add "userCanRead" as part of artifact type info?

Reply:
Added the logic for userCanRead